src/compiler/core.ts

```diff
@@ -661,6 +669,8 @@ namespace ts {
     /**
      * Compacts an array, removing any falsey elements.
      */
+    export function compact<T>(array: T[]): T[];
+    export function compact<T>(array: ReadonlyArray<T>): ReadonlyArray<T>;

```

Comment:
Is this overload order correct? Won't we just always pick the top one, since a `ReadonlyArray` will be compatible with it (so we'll never go down the list)? (Same comment on `singleOrMany`, `concatenate`, `sameFlatMap`, `sameMap`, and `filter`)